backends/dpdk/dpdkHelpers.cpp

```diff
@@ -51,9 +51,53 @@ void ConvertStatementToDpdk::process_relation_operation(const IR::Expression* ds
     add_instr(new IR::DpdkLabelStatement(end_label));
 }
 
+/* DPDK target does not support storing result of logical operations such as || and &&.
+   Hence we convert the expression var = a LOP b into if-else statements.
+   var = a || b            ||     var = a && b
+=> if (a){                 ||=>   if (!a)
+       var = 1;            ||         var = 0;
+   } else {                ||     } else {
+       if (b)              ||         if (!b)
+           var = 1;        ||             var = 0;
+       else                ||         else
+           var = 0;        ||             var = 1;
+   }                       ||     }
+
+   This function assumes complex logical operations are converted to simple expressions by
+   ConvertLogicalExpression pass.
+*/
+void ConvertStatementToDpdk::process_logical_operation(const IR::Expression* dst,
+                                                        const IR::Operation_Binary* op) {
+    if (!op->is<IR::LOr>() && !op->is<IR::LAnd>())
+        return;
+    auto true_label = Util::printf_format("label_%dtrue", next_label_id);
+    auto false_label = Util::printf_format("label_%dfalse", next_label_id);
+    auto end_label = Util::printf_format("label_%dend", next_label_id++);
+    if (op->is<IR::LOr>()) {
```

Comment:
I find this rather inefficient. On most architectures I know branches are actually significantly more expensive than Boolean operations. I expect you may want to reconsider this design decision.

Reply:
Yes. That's right. But in the absence of instructions to store the result of these operations or use them directly in jump instructions, I see no other way to handle these. We can revisit this if and when DPDK target introduces instructions for these logical operations.